website/src/theme/DocSidebarItem/index.js

```diff
@@ -12,18 +12,26 @@ import DocSidebarItemHtml from '@theme/DocSidebarItem/Html';
 import {usePluginData} from '@docusaurus/useGlobalData';
 import VersionContext from '../../stores/VersionContext'
 import pageVersionCheck from '../../utils/page-version-check';
+import categoryVersionCheck from '../../utils/category-version-check';
 
 export default function DocSidebarItem({item, ...props}) {
 
   // dbt Custom
-  const { versionedPages } = usePluginData('docusaurus-build-global-data-plugin');
+  const { versionedPages, versionedCategories } = usePluginData('docusaurus-build-global-data-plugin');
   const { version } = useContext(VersionContext)
   if(version && versionedPages) {
     const { pageAvailable } = pageVersionCheck(version, versionedPages, item.docId)
     if(!pageAvailable)
       return null
   }
 
+  // hide versionedCategories if they do not match the current version
+  if(version && versionedCategories) {
+    const { categoryAvailable } = categoryVersionCheck(version, versionedCategories, item.label)
+    if(!categoryAvailable)
+      return null
+  }
```

Comment:
Thinking we should add one additional check here to ensure the current `item` is of `type: category`. For example (although unlikely), if any document in the sidebar has the same label as the category, it would be hidden as well. 

I verified this is the case locally by updating a label for a random doc to the same name as the `Publishing models` category. If switching to v1.4 or lower, the doc would be hidden as well by the category version check.

Let me know if you have any concerns with adding the additional check if it could potentially effect anything else here.

![image](https://user-images.githubusercontent.com/13647110/229173035-06f9f68d-fc09-40ff-9617-475f6afc231d.png)
![image](https://user-images.githubusercontent.com/13647110/229172858-51fd2881-a492-46f9-b2aa-4a3d7e3444ba.png)

Reply:
Good point. I'll get get that check implemented!

Reply:
Added `if(version && versionedCategories && item.type === 'category')`

Let me know if you like that.

Reply:
![](https://media.giphy.com/media/3o6ZtpvPW6fqxkE1xu/giphy.gif)

Reply:
This all looks great to me. Thank you @john-rock and @JKarlavige.